src/DaprExtension/Services/DaprServiceClient.cs

```diff
@@ -112,6 +112,42 @@ static string GetDefaultDaprAddress(INameResolver resolver)
             return this.httpClient.SendAsync(req, cancellationToken);
         }
 
+        internal async Task<JObject> GetSecretAsync(
+            string? daprAddress,
+            string secretStoreName,
+            string? key,
+            string? metadata,
+            CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(secretStoreName))
+            {
+                throw new ArgumentNullException(nameof(secretStoreName));
+            }
+
+            if (string.IsNullOrEmpty(key))
```

Comment:
Why not just make the parameter non-nullable if we are going to throw an error at runtime anyways?

Reply:
I went back and forth on this and I think there are pros and cons to consider. The nice thing about having a nullable reference type here is that my error handling can be consolidated into one location. If I make this non-nullable, then I'm forcing the caller to do the null handling. If there are multiple callers then that means each caller has to do its own (potentially redundant) null handling. I went for the option which is less code, though again, I'm not sure what the best approach is.

Reply:
My hunch is that forcing the caller to handle the nullability question is the correct move, even if it may result in more code. The use of non-nullable types should be "viral", similar to how `async` is in C#. If the method requires non-null values, than the caller should also provide it a non-null value. 

This forces the caller to make the input it passes in non-nullable. If the caller is getting that input as a parameter itself where it is nullable, keep going up the call-chain making it non-nullable until you get to a point where the parameter is entirely non-nullable up the call-stack, or you reach a point where it makes sense for the value to be nullable, and perform error handling there.

This may sometime result in more code, but all null checks will be sure to be handled as close to where nullability is introduced, which I think is the intention. It also means that we are given compile time guarantees around null in most of our code, and only have to handle null argument exceptions on our input surface.

Note I am not an expert on this, but this is how `Option/Maybe` works in more pure functional languages, which is where I think the inspiration came from for the opt-in nullability.